plans/run.pp

```diff
@@ -0,0 +1,70 @@
+# Starts a Puppet agent run on the specified targets.
+# Note: This plan may cause issues when run in Puppet Enterprise.
+# @param targets The targets to start a Puppet agent run on.
+plan puppet_agent::run (
+  TargetSpec $targets
+) {
+  # Check which targets have the agent installed by checking
+  # the version of the agent. No point in trying to run the
+  # agent if it's not installed.
+  $version_results = run_task(
+    'puppet_agent::version',
+    $targets,
+    'Check for Puppet agent',
+    '_catch_errors' => true
+  )
+
+  # Create results with more descriptive error messages for any targets
+  # where the version task failed.
+  $fail_results = $version_results.error_set.map |$result| {
```

Comment:
I think `version_error_results` might be less confusing here, mostly since it's not clear what the difference is between a fail result and error result (plus we don't fail the plan here)

Reply:
Updated! Also changed `error_results` to `agentless_error_results` for similar reasons.